app/component/map/navigation-map.cjsx

```diff
@@ -0,0 +1,124 @@
+React         = require 'react'
+polyUtil      = require 'polyline-encoded'
+getSelector   = require '../../util/get-selector'
+geoUtils      = require '../../util/geo-utils'
+LocateActions = require '../../action/locate-actions.coffee'
+
+
+dataAsGeoJSON = (data) ->
+  res =
+    type: "FeatureCollection"
+    features: []
+
+  for leg in data.legs
+    res.features.push
+      type: "Feature"
+      geometry:
+        type: "LineString"
+        coordinates: ([i[1], i[0]] for i in polyUtil.decode leg.legGeometry.points)
+      properties:
+        mode: leg.mode.toLowerCase()
+  res
+
+locationAsGeoJSON = (coordinates) ->
+  type: "FeatureCollection"
+  features: [
+    type: "Feature"
+    geometry:
+      type: "Point"
+      coordinates: [coordinates.lon, coordinates.lat]
+  ]
+
+getLayerForMode = (mode) ->
+  id: mode
+  type: 'line'
+  source: 'route'
+  filter: ["==", "mode", mode]
+  layout:
+    "line-cap": 'round'
+  paint:
+    "line-width": 3
+    "line-color": getSelector(".#{mode}").style?.color or "#999"
+
+getLayerforLocation = ->
+  id: 'location'
+  type: 'symbol'
+  source: 'location'
+  layout:
+    "icon-allow-overlap": true
+    "icon-ignore-placement": true
+    "icon-image": "location"
+
+class NavigationMap extends React.Component
+  @contextTypes:
+    getStore: React.PropTypes.func.isRequired
+    executeAction: React.PropTypes.func.isRequired
+
+  componentDidMount: ->
+    @context.executeAction LocateActions.startLocationWatch
+
+    require.ensure ['mapbox-gl/dist/mapbox-gl-dev', './mapbox-gl.css'], =>
+      require './mapbox-gl.css'
```

Comment:
I changed the import logic quite a lot, now all of the requires are in the componentWillMount, where they are asynchronously required. The componentDidMount will then continue loading them after they have been downloaded.